4.1 AI agent chat (#670)
SimpleDataLabsInc/prophecy-docs: docs/analysts/development/ai-agent/ai-explore.md

This way, you can identify which dataset aligns better with your pipeline’s requirements, such as having the right fields, consistent naming conventions, or expected formats, without needing to inspect the full data.

## View sample rows from a table

To preview data, ask the agent to return sample rows from a table. You can ask to:

- Display a random sample from the table.
- Retrieve specific rows, such as “the ten most recent purchases over $100.”

The agent returns:

- A small table showing sample data directly in the chat.
- An option to add the full dataset as a Table gem in the pipeline canvas.
- An option to preview the chart more closely.

### Table preview

To inspect the sample data more closely, click **Preview**. Inside the preview, you can:

- Download the data as a JSON, Excel, or CSV file.
- Show or hide columns.
- Add the full dataset to the pipeline canvas.

## Visualize table data

You can also explore data through visualizations. Use prompts like “visualize sales by region over time” to generate charts.

The agent returns:

> - A chart embedded directly in the chat.

Author: query information available until the chart loads --- then, this information disappears

Author: is it only intended for data samples & visualizations? or also gem generation?

- An option to add the underlying dataset as a Table gem in the pipeline canvas.
- An option to preview the chart more closely.

### Chart preview

To see a larger version of the chart, click **Preview**. This opens the data visualization dialog, which has two tabs.

| Tab           | Available actions                                                                                                                                                                                                            |
| ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Visualization | <ul class="table-list"><li>View a larger version of the chart</li><li>Download the chart as an image</li><li>Copy the chart as an image</li><li>Add the chart to a [Prophecy App](/analysts/business-applications)</li></ul> |

Author: chart only available for target tables

Author: can't reproduce in beta ?

---
title: Explore data
id: ai-explore
slug: /analysts/ai-explore
description: Find data sources using Prophecy's AI agent
tags:
  - analyst
  - ai
  - agent
---

> One way to leverage the Prophecy AI agent is to search your SQL warehouse, explore datasets, and generate insights with simple prompts. This way, you can add the appropriate sources to your pipeline that will undergo data processing. You must have source data in your pipeline to start building transformations.

Reviewer: ```suggestion
One way to leverage the Prophecy AI agent is to search your SQL warehouse, explore datasets, and generate insights with simple prompts. This allows you to add the appropriate sources to your pipeline that will undergo data processing. You must have source data in your pipeline to start building transformations.
```
Nit: Wording.

> The following sections describe ways you can interact with the agent in this context.

Reviewer: ```suggestion
The following sections describe the ways you can interact with the agent.
```
Nit: Remove "in this context" since I think it's implied after reading the last paragraph.

:::note

> Agent responses are generated dynamically and may vary slightly from one request to another.

Reviewer: ```suggestion
Agent responses are dynamically generated and may vary slightly from one request to another.
```
Nit: "are generated dynamically" wasn't reading right to me.

:::

## Find tables in your SQL warehouse

> Ask the agent to search for tables in your [primary SQL warehouse](docs/administration/fabrics/prophecy-fabrics/prophecy-fabrics.md). Based on your query, it returns:

Reviewer: ```suggestion
You can ask the agent to search for tables in your [primary SQL warehouse](docs/administration/fabrics/prophecy-fabrics/prophecy-fabrics.md). Based on your query, the agent returns:
```
Nit: Wording.

Suggestion: Also, I think for the full list of datasets, it's worth mentioning you need to click "view full list" (or whatever the wording in the agent is) since users might not notice it.

- A short list of relevant datasets and their descriptions.
- A full list of datasets that match your criteria.

:::tip
You can add a dataset to the pipeline as a [Table gem](/analysts/table) directly from the chat.
:::

### Explore datasets

To learn more about a dataset, you can click on it inside the chat. This opens a dialog where you can:

- Preview a sample of the data in the dataset.
- Review the data profile of the sample.
- Find the location of the dataset in the warehouse.
- Look over the schema of the dataset.

> - Open the **Explore** AI agent that limits responses to the context of the dataset.

Reviewer: Question: Is "Explore AI Agent" its actual name? I thought it was the "Explore" window in the AI agent

This helps you quickly locate the correct data without having to manually browse through your data catalogs.

> ## Describe a dataset

Reviewer: Suggestion: For this and the next section, I think it's worth providing an example phrase since it's a bit buggy depending on how you word it -- like how a user should and should NOT word something. Or for this initial publish, at least a good example or two of how users should do it.

Author: I don't think we have any guidelines on what DOESN'T work, but I can provide a few working examples

If you want a summary of dataset information, you can ask the AI agent to describe a table.

The agent will provide a quick overview of key metadata, including the database and schema the table belongs to, as well as the names and data types of each column. This helps you understand the structure of the dataset without having to open or query it directly.

## Compare datasets

If you ask the AI agent to compare datasets, you can quickly assess which dataset is more suitable as a source for your pipeline by analyzing differences in schema structure, column names, data types, and size.

> This way, you can identify which dataset aligns better with your pipeline’s requirements, such as having the right fields, consistent naming conventions, or expected formats, without needing to inspect the full data.

Reviewer: ```suggestion
This allows you to identify which dataset aligns better with your pipeline’s requirements, such as having the right fields, consistent naming conventions, or expected formats, without needing to inspect the full data.
```

Nit: wording.

## View sample rows from a table

> To preview data from a table, ask the agent to return a sample. You can request a random sample from the table or specific rows, such as “the ten most recent purchases over $100”. The agent returns:

Reviewer: ```suggestion
To preview data from a table, ask the agent to return a sample. You can request a random sample from the table or specific rows, such as “the ten most recent purchases over $100”. 

The agent returns:
```

- A table showing the sample data
- A **Preview** option for a closer look
- SQL execution logs for transparency

If the dataset isn’t already part of your pipeline, you’ll also see an option to add it as a Table gem on the canvas.

![Data sample response](img/agent-sql-logs-sample.png)

### Table preview

> Click **Preview** to open a larger view of the data sample. In this view, you can:

Reviewer: ```suggestion
Click **Preview** to open a larger view of the data sample. In this view, you can:
```

Question/Suggestion: Where exactly is Preview? I looked at the screenshot above and below and didn't notice it in either. I think it's worth clarifying in the description here so users don't miss it too.

- An embedded chart directly in the chat
- A Preview button to open a detailed view
- SQL execution logs showing how the chart was generated

If the chart is based on a dataset that isn't yet in your pipeline, you'll see an option to add it as a Table gem.

![Chart generation response](img/agent-sql-logs-charts.png)

### Chart preview

To see a larger version of the chart, click **Preview**. This opens the data visualization dialog, which has two tabs.

| Tab           | Available actions                                                                                                                                  |
| ------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
| Visualization | <ul class="table-list"><li>View a larger version of the chart</li><li>Download the chart as an image</li><li>Copy the chart as an image</li></ul>  |
| Data          | <ul class="table-list"><li>View the underlying data</li><li>Download the data as a JSON, Excel, or CSV file</li><li>Show or hide columns</li></ul> |

:::info

> Learn more about data visualization in [Charts](/analysts/charts).

Reviewer: ```suggestion
To learn more about data visualization, see [Charts](/analysts/charts).
```

:::

## Sample prompts

> Here are some sample prompts that you can ask to search, explore, and learn about the data.

Reviewer: Ah, I see you did put prompts here, but I think they'd be more helpful at the end of their sections. OR, you can link to this at the end of each section :)
